### PR TITLE
Allow passing component attributes to preview generator

### PR DESCRIPTION
**Pull request by bobmaerten**
### Summary

This PR allows the preview generator to get the provided component generator attributes.
Why?
Because, the default generated preview when passing attributes is not valid and fails tests, and it's not cool to generate failing tests files. ;)

### Other Information

This PR move preview generator tests to its own file as for other generators.
A config option has been added too, in order to enable project-wide preview generation.

### CODE REVIEW

**Comment by bobmaerten** [comment]
I rebased this PR upon main, but still the test suite have a glitch on one version of rails with 2.5 which seems to have no link with the changes I made.

**Comment by Spone** [comment]
@bobmaerten re-running the tests fixed the glitch.

**Comment by bobmaerten** [comment]
👋🏻 Hello there 👋🏻 
Just rebase this PR upon 2.48.0 release.
Enjoy!

**Comment by BlakeWilliams** [comment]
@Spone Would you mind reviewing this PR when you get a chance? 🙇

**Comment by bobmaerten** [comment]
Hello again 👋🏻
I just rebased this PR upon 2.49.0 release.
Hope you are doing well folks.

**Comment by Spone** [comment]
Hi! I think we should wait for https://github.com/github/view_component/pull/1249 to be merged, in order to adapt this PR to the new syntax (`config.view_component.generate_preview` will become `config.view_component.generate.preview`).

**Comment by bobmaerten** [comment]
I shall rebase my work upon #1249 then.
I'll look into it, thanks.

**Comment by Spone** [comment]
@bobmaerten FYI #1249 has been merge, you can go ahead and rebase.

File: lib/rails/generators/component/component_generator.rb [hunk 1]
```diff
@@ -11,11 +11,13 @@ class ComponentGenerator < Rails::Generators::NamedBase
...
       check_class_collision suffix: "Component"
+
       class_option :inline, type: :boolean, default: false
+      class_option :locale, type: :boolean, default: ViewComponent::Base.generate.locale
       class_option :parent, type: :string, desc: "The parent class for the generated component"
-      class_option :stimulus, type: :boolean, default: ViewComponent::Base.generate.stimulus_controller
+      class_option :preview, type: :boolean, default: ViewComponent::Base.generate.preview
       class_option :sidecar, type: :boolean, default: false
-      class_option :locale, type: :boolean, default: ViewComponent::Base.generate.locale
+      class_option :stimulus, type: :boolean, default: ViewComponent::Base.generate.stimulus_controller
```
Comment (bobmaerten) [inline lib/rails/generators/component/component_generator.rb:20]: Juste sorting lines ;)

**Comment by Spone** [comment]
Looking good! Would you mind giving this a look @boardfish?

**Review by Spone — APPROVED** [review]

File: lib/rails/generators/preview/templates/component_preview.rb.tt [hunk 2]
```diff
@@ -1,5 +1,9 @@
 class <%= class_name %>ComponentPreview < ViewComponent::Preview
   def default
+  <%- if render_signature -%>
+    render(<%= class_name %>Component.new(<%= render_signature %>))
+  <%- else -%>
     render(<%= class_name %>Component.new)
+  <%- end -%>
```
Comment (boardfish) [inline lib/rails/generators/preview/templates/component_preview.rb.tt:7]:
```suggestion
  render(<%= class_name %>Component.new<%= "(#{render_signature})" if render_signature %>)
```

**Review by boardfish — APPROVED** [review]

File: lib/rails/generators/preview/templates/component_preview.rb.tt (hunk 2 above)
Comment (Spone) [reply lib/rails/generators/preview/templates/component_preview.rb.tt:7]: The indentation is needed though :)

File: lib/rails/generators/preview/templates/component_preview.rb.tt [hunk 3]
```diff
@@ -1,5 +1,5 @@
 class <%= class_name %>ComponentPreview < ViewComponent::Preview
   def default
-    render(<%= class_name %>Component.new)
+  render(<%= class_name %>Component.new<%= "(#{render_signature})" if render_signature %>)
```
Comment (Spone) [inline lib/rails/generators/preview/templates/component_preview.rb.tt:3]:
```suggestion
    render(<%= class_name %>Component.new<%= "(#{render_signature})" if render_signature %>)
```

Comment (boardfish) [reply lib/rails/generators/preview/templates/component_preview.rb.tt:3]: Oops - thanks for the save!

**Comment by bobmaerten** [comment]
Do you think adding `# frozen_string_literal: true` to generated code is a good idea/practise?

**Comment by boardfish** [comment]
I see no reason not to - it's always been something to lint for, and it was even debated about becoming a default in Ruby 3. Word has it Matz dropped it to maintain compatibility with existing code after upgrades, but I think it's relatively safe to say that generated components will be new code in new apps, which are likely to default to `frozen_string_literal: true`. We can always make that aspect configurable, but I don't think that'd be worthwhile unless it's requested.

**Comment by bobmaerten** [comment]
May be should we change it in another small quickreviewable PR, including all generated content at once, then.

**Pull request merged**